### PR TITLE
Gp kotlin conversions

### PR DESCRIPTION
**Pull request by mirahazall**
## Description

We've decided to restructure the codebase to use a single programming language for the whole project. Right now the project uses a combination of Java and Kotlin. Some files are written in Java and some are written in Kotlin. We’ve decided to transition entirely to Kotlin for consistency and maintainability. As part of this goal, I converted all files in the model folder including the CarePlan class from Java to Kotlin. I also converted all the Java files in the views → GP folder from Java to Kotlin. Those files include "GPAddFragment" and "GPProfileFragment". I also converted the files in the nextofkin folder. The files I converted in that folder are "NextOfKinFragment" and "NoKAddFragment".

Additionally, I upgraded the versions of our project dependencies to use the latest versions. 

### Todos

- [x] Tested and working locally
- [x] Code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] Code changes documented
- [x] Requested review from >= 2 devs on the team (one frontend and one backend recommended)

### How to test

The build on Android Studio was successful. 

### Screenshots and/or Gifs

[Insert screenshots and/or gifs showing the graphic representation of the change]

### Associated MS Planner Tasks

- [In views/GP, convert all Java to Kotlin](https://tasks.office.com/deakin365.onmicrosoft.com/Home/Task/4T-25yXsFUGifgww_HjM8cgAAvvX?Type=TaskLink&Channel=Link&CreatedTime=638598843743590000)
- [In views/nextofkin, convert all Java files to Kotlin](https://tasks.office.com/deakin365.onmicrosoft.com/Home/Task/L1AzR5KRhkSQ6PkCU_vxGMgAHbDA?Type=TaskLink&Channel=Link&CreatedTime=638598844327580000)
- [Convert all models from Java to Kotlin](https://tasks.office.com/deakin365.onmicrosoft.com/Home/Task/JB9NuQ812U6gxg0LYgNIT8gAGsLZ?Type=TaskLink&Channel=Link&CreatedTime=638598844822740000)

### Known Issues

Issue detected by spotless lint: Compatibility issues between java version and project dependency versions. According to the error, there is a mismatch between the Java version our Gradle build is configured to use and the Java version required by some of the dependencies, like google-services and the Android Gradle Plugin. These dependencies expect Java 11, but it says our project is set up for Java 8.

### CODE REVIEW

**Review by AsherAli22 — APPROVED** [review]

**Comment by AsherAli22** [comment]
Good Work @mirahazall

File: app/build.gradle
Comment (Onwa1kenobi) [inline app/build.gradle:1]: You have performed some version bumps here. Can you also mention this in the PR description?

**Review by Onwa1kenobi — COMMENTED** [review]
Very good work @mirahazall. Just need to actually fill out the PR template when creating a PR.

**Review by Onwa1kenobi — CHANGES_REQUESTED** [review]
The Git workflow build failed due to some code violations flagged by our spotless lint. Run `./gradlew :app:spotlessApply` to fix these violations, and then commit and push again.

File: app/build.gradle
Comment (mirahazall) [reply app/build.gradle:1]: Hi, I will mention the version modifications. I ran the ./gradlew :app:spotlessApply command and got an error that says there is a mismatch between the Java version our Gradle build is configured to use and the Java version required by some of the dependencies, like google-services and the Android Gradle Plugin. These dependencies expect Java 11, but our project is set up for Java 8. I'm not sure how to fix this incompatibility issue. In our build.gradle file, Java version is specified as 17. I'm not sure why spotless thinks we are using Java 8

Comment (Onwa1kenobi) [reply app/build.gradle:1]: That is weird. I will take a look.

Comment (Onwa1kenobi) [reply app/build.gradle:1]: Looks good now. You can Squash and Merge now @mirahazall.

**Review by Onwa1kenobi — APPROVED** [review]

**Pull request merged**